rpa-mapper/src/main/java/uk/gov/hmcts/cmc/rpa/mapper/DefendantJsonMapper.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.cmc.rpa.mapper;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.domain.models.otherparty.OrganisationDetails;
+import uk.gov.hmcts.cmc.domain.models.otherparty.SoleTraderDetails;
+import uk.gov.hmcts.cmc.domain.models.otherparty.TheirDetails;
+import uk.gov.hmcts.cmc.domain.models.party.HasContactPerson;
+import uk.gov.hmcts.cmc.domain.models.party.Party;
+import uk.gov.hmcts.cmc.rpa.mapper.helper.RPAMapperHelper;
+import uk.gov.hmcts.cmc.rpa.mapper.json.NullAwareJsonObjectBuilder;
+
+import java.util.List;
+import javax.json.JsonArray;
+import javax.json.JsonObjectBuilder;
+import javax.json.stream.JsonCollectors;
+
+import static uk.gov.hmcts.cmc.rpa.mapper.helper.Extractor.extractOptionalFromSubclass;
+
+@Component
+@SuppressWarnings({"LineLength"})
+public class DefendantJsonMapper {
```

Comment:
Please rename to `TheirDetailsJsonMapper` because `TheirDetails` might be used not only to represent defendants.

Reply:
I think it is used to fill in defendants key in json in both cases claim and defence. I think name is fine in this case.